cpp/SnowflakeGCSClient.cpp

```diff
@@ -66,6 +66,18 @@ RemoteStorageRequestOutcome SnowflakeGCSClient::upload(FileMetadata *fileMetadat
   if (!m_gcsAccessToken.empty())
   {
     std::string filePathFull = m_stageInfo->location + fileMetadata->destFileName;
+
+    //check if file exists if overwrite is not specified.
+    if (!fileMetadata->overWrite)
```

Comment:
Can we add condition, if it's not presigned URL with GCP. All other cloud provider is fine. i.e. condition like if(!overwrite && ((GCP && !presigned url) || AWS || AZURE)) then check for file exist.

Reply:
We have the correct condition already.
SnowflakeGCSClient.cpp is implementation for GCP only. It's under if `(!m_gcsAccessToken.empty())` which means not using presigned URL, so only need to add check for overwrite